tensorflow/python/compiler/tensorrt/trt_convert.py

```diff
@@ -111,70 +110,106 @@ def supported_precision_modes():
-# conversion.
-TrtConversionParams = collections.namedtuple(
-    "TrtConversionParams",
-    [
-
-        # A template RewriterConfig proto used to create a TRT-enabled
-        # RewriterConfig. If None, it will use a default one.
-        "rewriter_config_template",
-
-        # The maximum GPU temporary memory which the TRT engine can use at
-        # execution time. This corresponds to the 'workspaceSize' parameter of
-        # nvinfer1::IBuilder::setMaxWorkspaceSize().
-        "max_workspace_size_bytes",
-
-        # One of TrtPrecisionMode.supported_precision_modes().
-        "precision_mode",
-
-        # The minimum number of nodes required for a subgraph to be replaced by
-        # TRTEngineOp.
-        "minimum_segment_size",
-
-        # Whether to generate dynamic TRT ops which will build the TRT network
-        # and engine at run time.
-        # i.e. Since TensorRT version < 6.0 does not support dynamic dimensions
-        # other than the batch dimension, when the TensorFlow graph has a
-        # non-batch dimension of dynamic size, we would need to enable this
-        # option. This option should be set to True in TF 2.0.
-        "is_dynamic_op",
-
-        # Max number of cached TRT engines for dynamic TRT ops.
-        # Created TRT engines for a dynamic dimension are cached.
-        # This is the maximum number of engines that can be cached.
-        # If the number of cached engines is already at max but none of them
-        # supports the input shapes, the TRTEngineOp will fall back to run the
-        # original TF subgraph that corresponds to the TRTEngineOp.
-        "maximum_cached_engines",
-
-        # This argument is ignored if precision_mode is not INT8. If set to
-        # True, a calibration graph will be created to calibrate the missing
-        # ranges. The calibration graph must be converted to an inference graph
-        # by running calibration with calibrate(). If set to False, quantization
-        # nodes will be expected for every tensor in the graph (exlcuding those
-        # which will be fused). If a range is missing, an error will occur.
-        # Please note that accuracy may be negatively affected if there is a
-        # mismatch between which tensors TRT quantizes and which tensors were
-        # trained with fake quantization.
-        "use_calibration",
-
-        # Max size for the input batch.
-        # This parameter is only effective when is_dynamic_op=False which
-        # is not supported in TF 2.0.
-        "max_batch_size",
-    ])
-
-DEFAULT_TRT_CONVERSION_PARAMS = TrtConversionParams(
-    rewriter_config_template=None,
-    max_workspace_size_bytes=DEFAULT_TRT_MAX_WORKSPACE_SIZE_BYTES,
-    precision_mode=TrtPrecisionMode.FP32,
-    minimum_segment_size=3,
-    is_dynamic_op=True,
-    maximum_cached_engines=1,
-    use_calibration=True,
-    max_batch_size=1)
+
+@tf_export("experimental.tensorrt.ConversionParams", v1=[])
+class TrtConversionParams(object):
```

Comment:
Drive by comment: Can you change this to subclass `namedtuple`, similar to what is done [here](https://github.com/tensorflow/tensorflow/blob/52eb7b53a484c088a30e19d380e8efb64ea423e3/tensorflow/python/ops/parsing_config.py#L57)? That way this still remains a namedtuple and you don't have to reimplement part of the namedtuple interface.

Reply:
Let's fix this in a follow-up commit.  CC @DEKHTIARJonathan

Reply:
@sanjoy let me know if you need me to do anything specific :)
Would be nice if that could be merged by TF 2.2 cut